roles/setup_gitops/tasks/main.yml

```
@@ -0,0 +1,119 @@
+---
+- name: Get available CSVs
+  kubernetes.core.k8s_info:
+    api_version: operators.coreos.com/v1alpha1
+    kind: ClusterServiceVersion
+    namespace: "{{ sg_gitops_namespace }}"
+  register: _sg_current_csvs
+
+- name: Fail if gitops CSV is not present
+  vars:
+    csv_details: "resources[*].metadata.name"
+    csvs: "{{ _sg_current_csvs | json_query(csv_details) }}"
+  ansible.builtin.fail:
+    msg: "No CSV for openshift-gitops-operator are present"
+  when:
+    - (csvs | select('match', 'openshift-gitops-operator') | list | length) == 0
+
+- name: Mirror required images to local registry
+  ansible.builtin.include_tasks: local-registry.yml
+  when:
+    - sg_local_registry | default('') | length > 0
+
+- name: Patch ArgoCD repo server for Kustomize and PolicyGenerator
+  kubernetes.core.k8s:
+    definition:
+      apiVersion: argoproj.io/v1beta1
+      kind: ArgoCD
+      metadata:
+        name: openshift-gitops
+        namespace: "{{ sg_gitops_namespace }}"
+      spec:
+        kustomizeBuildOptions: "--enable-alpha-plugins"
+        repo:
+          volumes:
+            - name: kustomize
+              emptyDir: {}
+          initContainers:
+            - name: kustomize-plugin
+              image: "{{ (sg_gitops_ztp_repo + ':' + (sg_gitops_ztp_tag | default(sg_default_ztp_tag))) }}"
```

Already has a default:
```suggestion
              image: "{{ sg_gitops_ztp_repo + ':' + sg_gitops_ztp_tag }}"
```